src/Common/ErrorCodes.cpp

```diff
@@ -578,12 +579,57 @@ namespace ErrorCodes
 
     std::string_view getName(ErrorCode error_code)
     {
-        if (error_code >= END)
+        if (error_code < 0 || error_code >= END)
             return std::string_view();
         return error_codes_names.names[error_code];
     }
 
     ErrorCode end() { return END + 1; }
+
+    void increment(ErrorCode error_code, bool remote, const std::string & message, const std::string & stacktrace)
+    {
+        if (error_code >= end())
+        {
+            /// For everything outside the range, use END.
+            /// (end() is the pointer pass the end, while END is the last value that has an element in values array).
+            error_code = end() - 1;
+        }
+
+        ValuePair inc_value{
+            !remote,    /* local */
+            remote,     /* remote */
+            0,          /* error_time_ms */
+            message,    /* message */
+            stacktrace, /* stacktrace */
+        };
+        values[error_code].increment(inc_value);
+    }
+
+    /// ValuePair
```

Comment:
This comment is useless a little bit ...

Reply:
This is just for logic split, but I'm OK, let's remote it